test/lib/BedrockTester.cpp

```diff
@@ -552,35 +559,66 @@ bool BedrockTester::waitForStates(set<string> states, uint64_t timeoutUS)
         }
         usleep(100'000);
     }
+
+    // we timed out, let's get some debugging info
+    string stateString;
+    auto it = states.begin();
+    while (it != states.end()) {
```

Comment:
use `SComposeList(states)` to get a string from the set

Reply:
nice, thanks